sdk/openai/openai/test/public/assistants.spec.ts

```diff
@@ -0,0 +1,397 @@
+          const getCityNicknameFunctionDescription =
+            "Gets the nickname for a city, e.g. 'LA' for 'Los Angeles, CA'.";
+          const getCityNicknameTool = {
+            type: "function" as const,
+            function: {
+              name: getCityNicknameFunctionName,
+              description: getCityNicknameFunctionDescription,
+              parameters: {
+                type: "object",
+                properties: {
+                  city: {
+                    type: "string",
+                    description: "The city and state, e.g. San Francisco, CA",
+                  },
+                },
+              },
+            },
+          };
+
+          let favoriteCityCalled = false;
+          let nicknameCalled = false;
+          const getResolvedToolOutput = (toolCall: {
+            id: string;
+            function?: any;
+          }): { output: string } => {
+            const toolOutput = { tool_call_id: toolCall.id, output: "" };
+            if (toolCall["function"]) {
+              const functionCall = toolCall["function"];
+              const functionName = functionCall.name;
+              const functionArgs = JSON.parse(functionCall["arguments"] ?? {});
+              switch (functionName) {
+                case favoriteCityFunctionName:
+                  toolOutput.output = getFavoriteCity();
+                  favoriteCityCalled = true;
+                  break;
+                case getCityNicknameFunctionName:
+                  toolOutput.output = getCityNickname(functionArgs["city"]);
+                  nicknameCalled = true;
+                  break;
+                default:
+                  toolOutput.output = `Unknown function: ${functionName}`;
+                  break;
+              }
+            }
+
+            return toolOutput;
+          };
+
+          const instructions = `You are a helpful assistant. Use the provided functions to help answer questions.
+            Customize your responses to the user's preferences as much as possible and use friendly
+            nicknames for cities whenever possible.
+        `;
+          const functionAssistant = {
+            model: "gpt-4-1106-preview",
+            name: "JS SDK Test Assistant - Nickname",
+            instructions,
+            tools: [getUserFavoriteCityTool, getCityNicknameTool],
+          };
+          const assistant = await client.beta.assistants.create(functionAssistant);
+          assert.isNotNull(assistant.id);
+          const thread = await client.beta.threads.create();
+          assert.isNotNull(thread.id);
+          const content = "What's the nickname of my favorite city?";
+          const role = "user";
+          const message = await client.beta.threads.messages.create(thread.id, { role, content });
+          assert.isNotNull(message.id);
+          assert.equal(message.thread_id, thread.id);
+          let run = await client.beta.threads.runs.createAndPoll(
```

Comment:
should this be a const?

Reply:
It is reassigned in line 370 with `toolOutputs`